bulk-ingester/src/test/java/org/alfresco/hxi_connector/bulk_ingester/processor/BulkIngestionProcessorTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.hxi_connector.bulk_ingester.processor;
+
+import static java.lang.String.format;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import org.alfresco.elasticsearch.db.connector.model.AlfrescoNode;
+import org.alfresco.hxi_connector.bulk_ingester.event.NodePublisher;
+import org.alfresco.hxi_connector.bulk_ingester.exception.BulkIngesterRuntimeException;
+import org.alfresco.hxi_connector.bulk_ingester.processor.mapper.AlfrescoNodeMapper;
+import org.alfresco.hxi_connector.bulk_ingester.processor.model.Node;
+import org.alfresco.hxi_connector.bulk_ingester.repository.BulkIngesterNodeRepository;
+import org.alfresco.hxi_connector.bulk_ingester.repository.IdRange;
+import org.alfresco.hxi_connector.bulk_ingester.spring.ApplicationManager;
+
+@ExtendWith(MockitoExtension.class)
+@SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
+class BulkIngestionProcessorTest
+{
+    @Mock
+    private BulkIngesterNodeRepository bulkIngesterNodeRepository;
+    @Mock
+    private BulkIngesterConfig bulkIngesterConfig;
+    @Mock
+    private AlfrescoNodeMapper alfrescoNodeMapper;
+    @Mock
+    private NodePublisher nodePublisher;
+    @Mock
+    private ApplicationManager applicationManager;
+    @InjectMocks
+    private BulkIngestionProcessor bulkIngestionProcessor;
+
+    @Test
+    public void shouldPublishNodesFromGivenIdRangeAndShoutDownApplication()
```

Comment:
```suggestion
    public void shouldPublishNodesFromGivenIdRangeAndShutDownApplication()
```
Although I like the idea of "shouting down" the application too.

Reply:
Ok - scrap that. This is a real thing I've not heard of! :facepalm:

Reply:
Or maybe it's just a common typo :joy:

Reply:
fixed 😂